master/internal/db/postgres_trial.go

```diff
@@ -27,7 +27,12 @@ func AddTrial(ctx context.Context, trial *model.Trial, taskID model.TaskID) erro
 	}
 
 	err := Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
-		run, v2 := trial.ToRunAndTrialV2()
+		projectID, err := experimentsProjectID(ctx, trial.ExperimentID, tx)
+		if err != nil {
+			return err
+		}
+
+		run, v2 := trial.ToRunAndTrialV2(projectID)
```

Comment:
since these get called together everywhere in the code (except for the test), maybe have a helper which fuses them? i.e. `trial.ToRunAndTrialV2(ctx, tx, experimentId)`?